### PR TITLE
Work around MSBuild race condition

### PR DESCRIPTION
**Pull request by jaredpar**

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure, @davkean, @jasonmalinowski

**Comment by davkean** [comment]
Lgtm

Sent from my Windows Phone
________________________________

File: build/scripts/check-msbuild.ps1 [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
-        $m = [regex]::match($line, $pattern, [System.Text.RegularExpressions.RegexOptions]::IgnoreCase)
+        $m = [regex]::Match($line, $pattern, $ignoreCase)
         if ($m.Success) {
             $num = $m.Groups[1].Value
             $text = $m.Groups[2].Value
             $location = $line.Substring(0, $m.Index)
+
+            # MSBuild race condition.  Doesn't break build so don't promote to error for now
+            #
+            # https://github.com/dotnet/roslyn/issues/10116
```
Comment (jasonmalinowski) [inline build/scripts/check-msbuild.ps1]: Is this the bug that is tracking fixing the underlying issue to remove this, or is tracking the reason for this addition?

**Comment by jasonmalinowski** [comment]
:+1:

File: build/scripts/check-msbuild.ps1 (hunk 1 above)
Comment (jaredpar) [inline build/scripts/check-msbuild.ps1]: Both.  It's basically tracking properly dealing with this "hack" I'm about to check in.

**Pull request merged**